numpy/core/tests/test_multiarray.py

```diff
@@ -7578,9 +7582,11 @@ def test_to_int_scalar(self):
         # gh-9972 means that these aren't always the same
         int_funcs = (int, lambda x: x.__int__())
         for int_func in int_funcs:
-            assert_equal(int_func(np.array([1])), 1)
-            assert_equal(int_func(np.array([0])), 0)
-            assert_equal(int_func(np.array([[42]])), 42)
+            assert_equal(int_func(np.array(1)), 1)
+            assert_equal(int_func(np.array(0)), 0)
+            with assert_warns(DeprecationWarning):
```

Comment:
You need a with block for each function call, so that you can prove that _both_ warn